src/compiler/checker.ts

```diff
@@ -15712,6 +15712,21 @@ namespace ts {
             if (getModifierFlags(node) & ModifierFlags.Abstract && node.body) {
                 error(node, Diagnostics.Method_0_cannot_have_an_implementation_because_it_is_marked_abstract, declarationNameToString(node.name));
             }
+
+            // Is this the correct time to make assertions against the inheritance chain?
+            // Have all other methods been resolved?  Probably need to record that an override exists
+            // and perform the actual resolution later.
+            if (getModifierFlags(node) & ModifierFlags.Override) {
+              checkOverrideMethodDeclaration(node);
+            }
+        }
+
+        function checkOverrideMethodDeclaration(node: MethodDeclaration) {
+          forEachEnclosingClass(node, enclosingClass => {

```

Comment:
You'll want to only checking the containing class. The function `forEachEnclosingClass` is for dealing with *nesting* of classes, not inheritance chains